src/main/java/me/magnet/consultant/ServiceIdentifier.java

```diff
@@ -123,6 +138,7 @@ public String toString() {
 		getDatacenter().ifPresent(dc -> descriptors.add("dc=" + dc));
 		getHostName().ifPresent(host -> descriptors.add("host=" + host));
 		getInstance().ifPresent(instance -> descriptors.add("instance=" + instance));
+		descriptors.add("tags=[" + getTags().stream().collect(Collectors.joining(",")) + "]");
```

Comment:
Put an if statement around this, and remove the brackets. The brackets are likely to mess up other things like setting keys, and you should only show this if you actually defined tags.